### DISCUSSION TITLE
如何修改 routeComponents ？ 原本使用的是 React.lazy 方式，如何优雅替换？

### DISCUSSION THREAD

**Discussion by Lemonreds** [umijs/umi#11812] [Q&A] [answered]
如：
login: React.lazy(() => import(/* webpackChunkName: "umi__plugin-auth__Login" *‘/Login.tsx')),
想改成：
login: Login // 或者其他

**Comment by fz6m** [comment, marked as answer]
在 umi 4 中所有路由都是默认拆包，这样可以减少加载 js 体积，如果不希望懒加载，要换掉只能全部换掉：

```ts
  routeLoader: { moduleType: 'cjs' }
```

但这样会生成非常大的单文件产物，因为所有路由都没有分包汇总在一起了。

目前做不到手动配置哪些路由拆包哪些不拆包，只能全拆或全不拆。

**Reply by penny006** [reply]
我是umi3，想拆包，配了dynamicImport，路由配的dynamic动态导入，但还是打包1个文件umi.js。请教一下啥原因呢？
![路由配置](https://github.com/umijs/umi/assets/153806265/f0470af5-00b5-46b2-a0a0-2124d0efa2f0)

**Reply by fz6m** [reply]
umi 3 不维护了，建议升级到 umi 4
